src/api.rs

```diff
@@ -80,6 +90,40 @@ impl API {
     };
   }
 
+  fn login(){
+    // Get credentials
+    let mut username = match env::var("CLINK_USERNAME") {
+      Ok(user) => {
+        if user.chars().all(char::is_alphanumeric) {
+          user
+        } else {
+          panic!("CSH Username Invalid! Must only contain alphanumeric symbols");
+        }
+      },
+      Err(_) => get_current_username().unwrap().into_string().unwrap(),
+    };
+    // TODO: Sanitize username
+    println!("Please enter password for {}, or nothing to exit: ", username);
+    let password = read_password().unwrap();
+    // Start kinit, ready to get password from pipe
+    let process = match Command::new("sh")
+                                .arg("-c")
+                                .arg(String::from("kinit ") + &String::from(username.trim()) + &String::from("@CSH.RIT.EDU"))
+                                .stdin(Stdio::piped())
+                                .stdout(Stdio::piped())
+                                .spawn() {
+        Err(why) => panic!("couldn't spawn kinit: {}", why),
+        Ok(process) => process,
+    };
+
+    // Pipe in password
+    process.stdin.unwrap().write_all(password.as_bytes()).unwrap();
+
+    let mut s = String::new();
+    process.stdout.unwrap().read_to_string(&mut s).unwrap();
```

Comment:
why?

Reply:
Provides the password incorrect/non existent

Reply:
Right, but where do we put `s` after we've read into it?

Reply:
@Mstrodl The purpose of the `s` is to act as a string to read the result of the `kinit` into. Somehow, as a side effect, it's printing the result to the screen, and solving an issue where the kinit process "hangs(?)" or "loops(?)" forever. Either way, this needs to be understood. Something is weird, or broken, and we gotta find out what.